mono/metadata/class.c

```diff
@@ -427,16 +464,23 @@ mono_type_get_name_recurse (MonoType *type, GString *str, gboolean is_recursed,
 			else
 				g_string_append_c (str, '+');
 		} else if (*klass->name_space) {
-			g_string_append (str, klass->name_space);
+			GString *esc_ns = mono_identifier_escape_type_name_chars (klass->name_space);
+			g_string_append (str, esc_ns->str);
 			g_string_append_c (str, '.');
+			g_string_free(esc_ns, TRUE);
```

Comment:
Pass the GString 'str' to mono_identifier_escape_type_name_chars. This avoids the intermediary GString.